go.mod

```diff
@@ -5,27 +5,27 @@ go 1.18
 require (
 	github.com/google/go-cmp v0.5.9
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.0
-	google.golang.org/grpc v1.51.0
-	google.golang.org/protobuf v1.28.1
+	google.golang.org/grpc v1.56.3
+	google.golang.org/protobuf v1.30.0
 	sigs.k8s.io/controller-runtime v0.14.1
 )
 
 require (
 	github.com/go-logr/logr v1.2.3 // indirect
 	github.com/go-logr/zapr v1.2.3 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/golang/protobuf v1.5.3 // indirect
 	github.com/google/gofuzz v1.1.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	go.uber.org/atomic v1.7.0 // indirect
 	go.uber.org/multierr v1.6.0 // indirect
 	go.uber.org/zap v1.24.0 // indirect
-	golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10 // indirect
-	golang.org/x/sys v0.3.0 // indirect
-	golang.org/x/text v0.5.0 // indirect
-	google.golang.org/genproto v0.0.0-20221207170731-23e4bf6bdc37 // indirect
+	golang.org/x/net v0.17.0 // indirect
```

Comment:
Is there a specific reason to use replace instead go get?

imho `go get` is better since it will also update go.sum entries and update related dependencies as well.
And, i guess, replace is more used for development purposes.

Reply:
Well, I suppose that there is some developer preference involved here, but in general, the idea is consistency and reproducibility from environment to environment (development, production, CI/CD, Docker builds). We really don't want to use `go get ...` unless we deliberately want to add new dependencies, or update individual dependencies, which will cause the `go.mod` and `go.sum` file to be updated. 

https://stackoverflow.com/questions/66356034/what-is-the-difference-between-go-get-command-and-go-mod-download-command

When I set up a Go project, meaning, I fork a GitHub repo and clone it locally, I will then run `go mod download` to get the dependencies with the exact versions as specified, or `go mod tidy -compat=1.18` (for a Go 1.18 project). The latter should not change the `go.mod` or `go.sum` file either, if the the two files have been kept in order.

I expect only the direct dependencies to be maintained. The list of `// indirect` dependencies should be "generated" by running `go mod tidy` (or `go mod tidy -compat=1.18` -- for a Go 1.18 project) which will determine the versions of those indirect dependencies based on the versions required by the modules from the list of "direct" requirements.

Now, how should we "override" certain indirect dependencies in the `go.mod` file? I think we should not just change a version of a indirect dependency, i.e. by running `go get ...` which will  to change the list `// indirect` dependencies. This would make it hard to determine in the future which versions we decided to use, versus what versions were required indirectly.

To keep that distinction clear, Go offers the `replace` directive :-)